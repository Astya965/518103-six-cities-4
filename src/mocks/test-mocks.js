import {CITIES, SortType} from "../utils/constants.js";

export const testOffers = [
  {
    id: `1`,
    isPremium: true,
    isFavourite: true,
    previewImage: `img/apartment-01.jpg`,
    price: 55,
    rating: 4.8,
    type: `Apartment`,
    title: `Test name-1`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    bedrooms: 2,
    maxAdults: 3,
    goods: [`Heating`, `Kitchen`, `Frigde`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatarUrl: `img/avatar-max.jpg`,
      id: `host-1`,
      isPro: true,
      name: `Maximum`,
    },
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`, `The building is green and from 18th century.`],
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
    },
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      },
      name: `Amsterdam`
    }
  },
  {
    id: `2`,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-02.jpg`,
    price: 250,
    rating: 2,
    type: `Room`,
    title: `Test name-2`,
    images: [`img/apartment-02.jpg`],
    bedrooms: 1,
    maxAdults: 1,
    goods: [`Heating`, `Kitchen`, `Frigde`],
    host: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: `host-3`,
      isPro: false,
      name: `Ellen`,
    },
    description: [`The house itself is surrounded by a gorgeous garden with many hidden lights that make the garden come to life at night.`],
    location: {
      latitude: 52.369553943508,
      longitude: 4.929309666406198,
    },
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      },
      name: `Amsterdam`
    }
  },
  {
    id: `3`,
    isPremium: true,
    isFavourite: true,
    previewImage: `img/apartment-02.jpg`,
    price: 155,
    rating: 3.2,
    type: `Hotel`,
    title: `Test name-3`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    bedrooms: 3,
    maxAdults: 4,
    goods: [`Heating`, `Kitchen`, `Frigde`, `Washing machine`, `Dishwasher`],
    host: {
      avatarUrl: `img/avatar-max.jpg`,
      id: `host-1`,
      isPro: true,
      name: `Maximum`,
    },
    description: [`Everything was fantastic. Location is excellent, very friendly staff, cleanliness flawless. I would definitely recommend this hotel to anyone.`],
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
    },
    city: {
      location: {
        latitude: 33.66094,
        longitude: -95.55551,
        zoom: 12
      },
      name: `Paris`,
    }
  },
];

export const testOffer = {
  id: `1`,
  isPremium: true,
  isFavourite: true,
  previewImage: `img/apartment-01.jpg`,
  price: 55,
  rating: 4.8,
  type: `Apartment`,
  title: `Test name-1`,
  images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
  bedrooms: 2,
  maxAdults: 3,
  goods: [`Heating`, `Kitchen`, `Frigde`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: {
    avatarUrl: `img/avatar-max.jpg`,
    id: `host-1`,
    isPro: true,
    name: `Maximum`,
  },
  description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`, `The building is green and from 18th century.`],
  location: {
    latitude: 52.3909553943508,
    longitude: 4.85309666406198,
  },
  city: {
    location: {
      latitude: 52.38333,
      longitude: 4.9,
      zoom: 12
    },
    name: `Amsterdam`
  }
};

export const testReviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: 1272402600000,
    id: `1`,
    rating: 5,
    user: {
      avatarUrl: `img/avatar-max.jpg`,
      id: `host-3`,
      isPro: false,
      name: `Leo`,
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: 1472402600000,
    id: `2`,
    rating: 3,
    user: {
      avatarUrl: `img/avatar-max.jpg`,
      id: `host-4`,
      isPro: false,
      name: `Melon`,
    }
  },
  {
    comment: `Didn't like that the bathroom door inside the room is wide open.
    Overall the hotel is really close to the metro/train station and it was clean and comfortable.`,
    date: 1492402600000,
    id: `3`,
    rating: 0,
    user: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: `host-5`,
      isPro: true,
      name: `Su`,
    }
  },
];

export const testReview = {
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: 1272402600000,
  id: `1`,
  rating: 5,
  user: {
    avatarUrl: `img/avatar-max.jpg`,
    id: `host-3`,
    isPro: false,
    name: `Leo`,
  }
};

export const testStore = {
  CITIES: {
    cities: CITIES,
    currentCity: CITIES[0],
  },
  OFFERS: {
    offers: testOffers,
    currentOffer: null,
    currentSort: SortType.DEFAULT,
  },
  REVIEWS: {
    reviews: testReviews
  }
};

export const testStoreWithCurrentOffer = {
  CITIES: {
    cities: CITIES,
    currentCity: CITIES[3],
  },
  OFFERS: {
    offers: testOffers,
    currentOffer: testOffer,
    currentSort: SortType.DEFAULT,
  },
  REVIEWS: {
    reviews: testReviews
  }
};
