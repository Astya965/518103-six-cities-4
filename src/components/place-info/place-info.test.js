import React from "react";
import renderer from "react-test-renderer";

import PlaceInfo from "./place-info.jsx";
import {testOffer} from "../../mocks/test-mocks.js";

it(`Render PlaceInfo`, () => {
  const tree = renderer.
    create(
        <PlaceInfo offer={testOffer}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
