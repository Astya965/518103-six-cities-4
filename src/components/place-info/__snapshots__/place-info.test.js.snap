// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PlaceInfo 1`] = `
Array [
  <div
    className="property__mark"
  >
    <span>
      Premium
    </span>
  </div>,
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      Test name-1
    </h1>
    <button
      className="property__bookmark-button button property__bookmark-button--active"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "96%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      4.8
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Apartment
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      2
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      3
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      55
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Heating
      </li>
      <li
        className="property__inside-item"
      >
        Kitchen
      </li>
      <li
        className="property__inside-item"
      >
        Frigde
      </li>
      <li
        className="property__inside-item"
      >
        Cable TV
      </li>
      <li
        className="property__inside-item"
      >
        Washing machine
      </li>
      <li
        className="property__inside-item"
      >
        Coffee machine
      </li>
      <li
        className="property__inside-item"
      >
        Dishwasher
      </li>
    </ul>
  </div>,
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
      >
        <img
          alt="Host avatar"
          className="property__avatar user__avatar"
          height="74"
          src="img/avatar-max.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Maximum
      </span>
    </div>
  </div>,
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
    </p>
    <p
      className="property__text"
    >
      The building is green and from 18th century.
    </p>
  </div>,
]
`;
